### PR TITLE
fix(kv_store): allow buffering io for key insert

### PR DESCRIPTION
**Pull request by Integralist**
The following code demonstrates a working example...

```go
package main

import (
	"fmt"
	"log"
	"os"
	"strings"

	"github.com/fastly/go-fastly/v8/fastly"
)

func main() {
	client, err := fastly.NewClient(os.Getenv("FASTLY_API_KEY"))
	if err != nil {
		log.Fatal("failed to create NewClient: ", err)
	}

	id := "<REDACTED>"
	key := "gofastly-testing-kvstore-"
	key1 := key + "1"
	key2 := key + "2"
	key3 := key + "3"

	err = client.InsertKVStoreKey(&fastly.InsertKVStoreKeyInput{
		ID:    id,
		Key:   key1,
		Value: "some value that is small",
	})
	if err != nil {
		log.Fatal("failed to insert key 1: ", err)
	}

	body := "some larger value, e.g. a file"
	err = client.InsertKVStoreKey(&fastly.InsertKVStoreKeyInput{
		Body: strings.NewReader(body),
		ID:   id,
		Key:  key2,
	})
	if err != nil {
		log.Fatal("failed to insert key 2: ", err)
	}

	// The file is ~3.3MB
	largeFile, err := os.Open("large.txt")
	if err != nil {
		log.Fatal("failed to open large.txt", err)
	}

	lr, err := fastly.FileLengthReader(largeFile)
	if err != nil {
		log.Fatal("failed to create length reader", err)
	}

	err = client.InsertKVStoreKey(&fastly.InsertKVStoreKeyInput{
		Body: lr,
		ID:   id,
		Key:  key3,
	})
	if err != nil {
		log.Fatal("failed to insert key 3: ", err)
	}

	v, err := client.GetKVStoreKey(&fastly.GetKVStoreKeyInput{
		ID:  id,
		Key: key1,
	})
	if err != nil {
		log.Fatal("failed to get key 1: ", err)
	}

	fmt.Printf("1. %+v\n", v)

	v, err = client.GetKVStoreKey(&fastly.GetKVStoreKeyInput{
		ID:  id,
		Key: key2,
	})
	if err != nil {
		log.Fatal("failed to get key 2: ", err)
	}

	fmt.Printf("2. %+v\n", v)

	v, err = client.GetKVStoreKey(&fastly.GetKVStoreKeyInput{
		ID:  id,
		Key: key3,
	})
	if err != nil {
		log.Fatal("failed to get key 3: ", err)
	}

	fmt.Printf("3. %+v\n", v)
}
```

### CODE REVIEW

**Comment by awilliams-fastly** [comment]
I'm seeing this output when running the above code:
```
2023/05/11 11:27:38 Using store "test-1"
Using request Value
Using request Body
Using request Body
2023/05/11 11:27:39 failed to insert key 3: 400 - Bad Request:

    Title:  Bad request
    Detail: must include valid content-length header
exit status 1
```

So it's failing when reading from a file, but not from `strings.Reader`. That seems odd at first, but I realized that Go's `http` package has an [optimization](https://go.dev/src/net/http/request.go#L899) where it'll automatically set the `Content-Length` header if the body is of type `strings.Reader` (among others). 

In the case of an `os.File`, we can call `Stat()` then `Size()` to get the file's size. Maybe this could be used somehow to set the `Content-Length` request header?

**Comment by Integralist** [comment]
Thanks @awilliams-fastly your suggestion worked great!

**Comment by Integralist** [comment]
@awilliams-fastly @dennismartensson I've integrated these go-fastly changes locally in the Fastly CLI and they work great 👍🏻

File: fastly/kv_store.go [hunk 1]
```diff
@@ -348,13 +349,49 @@ func (c *Client) GetKVStoreKey(i *GetKVStoreKeyInput) (string, error) {
...
+	// Body is the value to insert and will be streamed to the endpoint.
+	// This is for users who are passing very large files.
+	// Otherwise use the 'Value' field instead.
+	Body LengthReader
 	// ID is the ID of the kv store (required).
 	ID string
 	// Key is the key to add (required).
 	Key string
-	// Value is the value to insert (required).
+	// Value is the value to insert.
```
Comment (awilliams-fastly) [inline fastly/kv_store.go:394]: Consider mentioning that this is ignored if `Body` is set.

**Review by awilliams-fastly — APPROVED** [review]

**Pull request merged**